### PR TITLE
Option to set metrics/log API log levels

### PR DESCRIPTION
**Pull request by matrixik**

### CODE REVIEW

File: chef/data_bags/crowbar/template-monasca.json [hunk 1]
```diff
@@ -38,12 +38,18 @@
...
       },
-      "api" : {
-        "bind_host" : "*",
-        "bind_port" : 8070,
-        "processes" : 3,
-        "ssl" : false,
-        "threads" : 10
+      "api": {
+        "bind_host": "*",
+        "bind_port": 8070,
```
Comment (dirkmueller) [inline chef/data_bags/crowbar/template-monasca.json]: not sure if there is a lot of value in making that configurable..

File: chef/data_bags/crowbar/template-monasca.json [hunk 2]
```diff
@@ -38,12 +38,18 @@
...
+        "processes": 3,
+        "ssl": false,
+        "threads": 10,
+        "log_level": "INFO"
+      },
+      "log_api": {
+        "port": 5607,
```
Comment (dirkmueller) [inline chef/data_bags/crowbar/template-monasca.json]: same here

File: chef/data_bags/crowbar/template-monasca.json [hunk 3]
```diff
@@ -38,12 +38,18 @@
...
+        "ssl": false,
```
Comment (dirkmueller) [inline chef/data_bags/crowbar/template-monasca.json]: we might want to do that as a sub dict with the usual settings going forward. for now I would not want to worry about it

File: crowbar_framework/app/views/barclamp/monasca/_edit_attributes.html.haml [hunk 4]
```diff
@@ -6,19 +6,32 @@
...
-      = password_field %w(master database_thresh_password)
-      = password_field %w(master database_logapi_password)
-      = password_field %w(master keystone_cmm_operator_user_password)
-      = password_field %w(master keystone_cmm_agent_password)
-      = password_field %w(master keystone_admin_agent_password)
-      = password_field %w(master database_grafana_password)
+        = t(".metric_agent_header")
+
+      = boolean_field %w(metric_agent system_only)
+      = string_field %w(metric_agent log_dir)
```
Comment (dirkmueller) [inline crowbar_framework/app/views/barclamp/monasca/_edit_attributes.html.haml]: should not be user visible imho

File: crowbar_framework/app/views/barclamp/monasca/_edit_attributes.html.haml [hunk 5]
```diff
@@ -6,19 +6,32 @@
...
+      = select_field %w(metric_agent log_level), :collection => :metric_agent_log_levels
+      = integer_field %w(metric_agent statsd_port)
```
Comment (dirkmueller) [inline crowbar_framework/app/views/barclamp/monasca/_edit_attributes.html.haml]: same here

**Review by dirkmueller — CHANGES_REQUESTED** [review]
LGTM, sorry for causing extra work here on your side. I did take a backup locally and had the mental todo to clean this up today. some comments inline. 

In general our commit messages are in the syle of "$component: headline change"  - the git commit summary is customer visible so we enforce a common style here. 

with nitpicks resolved this is ready to be merged then (could you send it to the crowbar repo then right away to not have this lost anymore?)

File: crowbar_framework/app/views/barclamp/monasca/_edit_attributes.html.haml (hunk 5 above)
Comment (artur-ba) [reply crowbar_framework/app/views/barclamp/monasca/_edit_attributes.html.haml]: @dirkmueller I think user should have possibility to configure statsd port for monasca-agent. We had a problem on Openstack installation where other service is using this port (for example this was a gnocchi server which as default also uses 8125 port as default). If port is already used agent won't start. So I think user should be able to configure this before installation.

File: chef/data_bags/crowbar/template-monasca.json (hunk 1 above)
Comment (matrixik) [reply chef/data_bags/crowbar/template-monasca.json]: Removed from here and added to chef/cookbooks/monasca/attributes/default.rb

File: chef/data_bags/crowbar/template-monasca.json (hunk 2 above)
Comment (matrixik) [reply chef/data_bags/crowbar/template-monasca.json]: Removed from here and added to chef/cookbooks/monasca/attributes/default.rb

File: chef/data_bags/crowbar/template-monasca.json (hunk 3 above)
Comment (matrixik) [reply chef/data_bags/crowbar/template-monasca.json]: removed

File: crowbar_framework/app/views/barclamp/monasca/_edit_attributes.html.haml (hunk 4 above)
Comment (matrixik) [reply crowbar_framework/app/views/barclamp/monasca/_edit_attributes.html.haml]: removed

File: crowbar_framework/app/views/barclamp/monasca/_edit_attributes.html.haml (hunk 5 above)
Comment (matrixik) [reply crowbar_framework/app/views/barclamp/monasca/_edit_attributes.html.haml]: left configurable

**Comment by matrixik** [comment]
Already merged with crowbar-openstack in https://github.com/crowbar/crowbar-openstack/pull/916

**Pull request closed without merging**